packages/validator/src/services/indices.ts

```diff
@@ -110,6 +116,49 @@ export class IndicesService {
         newIndices.push(validatorState.index);
       }
     }
+
+    logValidatorStatuses(allValidators, this.logger);
+
     return newIndices;
   }
 }
+
+function logValidatorStatuses(allValidators: Map<ValidatorStatus, ValidatorResponse[]>, logger: Logger): void {
+  const loggedValidatorStates: {[key: string]: ValidatorStatus[]} = {
+    "Validator activated": ["active"],
+    "Validator pending": ["pending_initialized", "pending_queued"],
+    "Validator withdrawn": ["withdrawal_done"],
+    "Validator exited": ["exited_slashed", "exited_unslashed"],
+  };
+
+  const logFormatter = (message: string, validatorState: ValidatorResponse): void => {
+    logger.info(message, {
+      pubkey: toHexString(validatorState.validator.pubkey),
+      index: validatorState.index,
+    });
+  };
+
+  for (const [message, statuses] of Object.entries(loggedValidatorStates)) {
+    getValidatorStatesByStatuses(statuses, allValidators).forEach((validatorState) => {
```

Comment:
project coding preferences thing: we try to avoid `x.forEach()` and instead do `for (const y of x) {`